src/routes/living-manage-offer/containers/LivingManageOfferPage.js

```diff
@@ -1,81 +1,70 @@
-/* eslint-disable */
 import React from 'react'
 import PropTypes from 'prop-types'
+import serverConfig from 'server.config'
+import getCurrentCityConfig from '../../../modules/city-detection/getCurrentCityConfig'
+import DeleteOfferManager from '../components/DeleteOfferManager'
+import Failure from '../../../modules/common/components/Failure'
+import ExtendOfferManager from '../components/ExtendOfferManager'
+import ConfirmOfferManager from '../components/ConfirmOfferManager'
 import { connect } from 'react-redux'
-import compose from 'lodash/fp/compose'
-import { Caption } from '@integreat-app/shared'
 
-const NOT_STARTED = 'notstarted'
-const DONE = 'done'
-const FAILED = 'failed'
-const PENDING = 'pending'
+const cityConfig = getCurrentCityConfig()
+
+const STATUS_OK = 200
 
 export class LivingManageOfferPage extends React.Component {
   static propTypes = {
     token: PropTypes.string.isRequired,
     action: PropTypes.string.isRequired
   }
 
-  constructor () {
-    super()
-    this.state = {success: NOT_STARTED}
-  }
-
-  componentWillMount () {
-    if (this.props.action === 'confirm') {
-      this.send('POST', `/confirm`)
-    }
-  }
+  state = {sending: false, success: false, serverError: null}
 
-  send (method, action = '') {
-    this.setState({success: PENDING})
-    fetch(`http://server11.integreat-app.de:8080/v0/${this.props.location}/${this.props.token}${action}`, {
-      method
-    }).then(response => {
-      console.log(response)
-      return response
+  send = (method, action = '', body = null) => {
+    this.setState({sending: true, serverError: null, success: false})
+    fetch(`${serverConfig.host}/v0/${cityConfig.cmsName}/${this.props.token}${action}`, {
+      method,
+      headers: new Headers({
+        'Content-Type': 'application/json'
+      }),
+      body: body && JSON.stringify(body)
     })
-      .then(response => response.status)
-      .then(status => this.setState({success: status === 200 ? DONE : FAILED}))
+      .then(response => {
+        if (response.status === STATUS_OK) {
+          this.setState({success: response.status === STATUS_OK, sending: false})
+        } else {
+          response.text().then(text => this.setState({success: false, sending: false, serverError: text}))
```

Comment:
Should we expose the error to the user?

Reply:
Not completely sure, how we should handle the different errors coming from the backend... For now I think it's better to expose more information about the error 🤔